### PR TITLE
Cknieling/support dependent attributes

### PR DESCRIPTION
**Pull request by chr11115**
- generate dynamic default values that depend on user set attributes
- add functionality to query dynamic default values for enums 
- minor improvements for setting attribute values
This PR depends on changes made in [PR 103](https://github.com/Esri/serlio/pull/103) and [PR 104](https://github.com/Esri/serlio/pull/104), therefore these should be reviewed first. 
The first actual commit for this PR is b9b29f3

### CODE REVIEW

**Review by mistafunk — CHANGES_REQUESTED** [review]
Applying our favela.rpk example to a 10x10 quad (switch scene to meters first), now produces this warning:

[error] maya status error at .../serlio/src/serlio/modifiers/PRTModifierAction.cpp:896: (kFailure): Object does not exist (code 1)

the model is still generated but it seems this PR introduces some unexpected side-effects

File: src/serlio/modifiers/PRTModifierAction.cpp [hunk 1]
```diff
@@ -61,7 +61,7 @@ const AttributeMapUPtr
 
 AttributeMapUPtr getDefaultAttributeValues(const std::wstring& ruleFile, const std::wstring& startRule,
                                            const prt::ResolveMap& resolveMap, prt::CacheObject& cache,
-                                           const PRTMesh& prtMesh, const int32_t seed) {
+                                           const PRTMesh& prtMesh, const int32_t seed, const AttributeMapUPtr& attributeMap) {
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierAction.cpp]: fixed

File: src/serlio/modifiers/PRTModifierAction.cpp [hunk 2]
```diff
@@ -237,6 +237,53 @@ MStatus addHiddenBoolParameter(MFnDependencyNode& node, MFnAttribute& tAttr, con
 	}
 	return stat;
 }
+
+short getDefaultEnumValue(const AttributeMapUPtr& defaultAttributeValues, const MFnEnumAttribute& eAttr,
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierAction.cpp]: fixed

File: src/serlio/modifiers/PRTModifierNode.cpp [hunk 3]
```diff
@@ -94,6 +94,8 @@ MStatus PRTModifierNode::compute(const MPlug& plug, MDataBlock& data) {
 			MDataHandle currentRulePkgData = data.inputValue(currentRulePkg, &status);
 			MCheckStatus(status, "ERROR getting currentRulePkg");
 
+			const bool ruleFileWasChanged = rulePkgData.asString() != currentRulePkgData.asString();
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierNode.cpp]: fixed

File: src/serlio/modifiers/PRTModifierAction.cpp [hunk 4]
```diff
@@ -237,6 +237,53 @@ MStatus addHiddenBoolParameter(MFnDependencyNode& node, MFnAttribute& tAttr, con
...
+					return currIdx;
+			}
+			break;
+		}
+		case prt::AAT_BOOL: {
+			const auto defBoolVal = defaultAttributeValues->getBool(fqAttrName.c_str());
+
+			for (short currIdx = minVal; currIdx <= maxVal; currIdx++) {
+				const int currBool = eAttr.fieldName(currIdx).asInt() != 0;
+				if (currBool == defBoolVal)
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierAction.cpp:277]: fixed, using `const bool currbool now`

**Comment by chr11115** [comment]
> Applying our favela.rpk example to a 10x10 quad (switch scene to meters first), now produces this warning:
> 
> [error] maya status error at .../serlio/src/serlio/modifiers/PRTModifierAction.cpp:896: (kFailure): Object does not exist (code 1)
> 
> the model is still generated but it seems this PR introduces some unexpected side-effects

Nice find, I will investigate the cause.

**Review by mistafunk — APPROVED** [review]
LGTM, I'll provide further cleanup suggestion in a separate PR

**Pull request merged**